x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeListener.java

```diff
@@ -188,6 +189,27 @@ ActionListener<Void> acquireAvoid() {
         });
     }
 
+    ActionListener<Void> acquireSkipUnavailable(@Nullable String computeClusterAlias) {
+        if (computeClusterAlias == null
+            || esqlExecutionInfo.isCrossClusterSearch() == false
+            || runningOnRemoteCluster()
+            || computeClusterAlias.equals(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY)
+            || esqlExecutionInfo.isSkipUnavailable(computeClusterAlias) == false) {
+            return acquireAvoid();
+        }
+        return refs.acquire().delegateResponse((l, e) -> {
+            LOGGER.error("Skipping unavailable cluster {} in ESQL query: {}", computeClusterAlias, e);
+            esqlExecutionInfo.swapCluster(computeClusterAlias, (k, v) -> {
+                assert v.getStatus() != EsqlExecutionInfo.Cluster.Status.SKIPPED
+                    : "We shouldn't be running compute on a cluster that's already marked as skipped";
+                return new EsqlExecutionInfo.Cluster.Builder(v).setStatus(EsqlExecutionInfo.Cluster.Status.PARTIAL)
```

Comment:
Can you put a code comment here as to why it is PARTIAL? (Namely, that some data may have already been sent down the exchanges and collected by the coordinators, so we always just assume this is a partial data situation.)